src/io.js

```diff
@@ -1,18 +1,27 @@
 import fs from "fs";
 import path from "path";
 import isUrl from "is-url";
+import { parseFile } from "./parser.js";
 
 async function getFromUrlOrFile(location, cache, base = null) {
   if (isUrl(location)) {
     return await cache.fetch(location);
   } else {
     if (base != null) {
-      return JSON.parse(
-        await fs.promises.readFile(path.join(base, location), "utf8"),
-      );
+      if (location.endsWith(".yml") || location.endsWith(".yaml"))
+        return parseFile(
```

Comment:
Calling `parseFile()` here is not really buying us anything over just invoking `yaml.load()` directly if we have to decide the format first. I might factor out a `parseSchema()` helper with a more restrictive set of formats, but we don't need to do it in this PR.

Reply:
Changed to `yaml.load`.